tidy3d/constants.py

```diff
@@ -13,20 +13,22 @@
 
 import numpy as np
 
+# floating point types and precisions
+TIDY3D_DOUBLE_PRECISION = False
+fp_type = np.float64 if TIDY3D_DOUBLE_PRECISION else np.float32
+cfp_type = np.complex128 if TIDY3D_DOUBLE_PRECISION else np.complex64
+fp_eps = np.finfo(fp_type).eps
```

Comment:
The frontend uses double precision (or more precisely, whatever precision numpy installation you have, which should almost always be double) almost everywhere. We have been using the single precision epsilon as a shorthand for "a small number" in a few places, even when the operations involved are actually double precision. In that sense I am not sure that using the actual precision epsilon would be sufficient in those places, and maybe we need to replace the current usage of `fp_eps` in the code with some `SMALL_NUMBER = 1e-8` type of thing.